### PR TITLE
Small changes with Jest Mocking and SEO Bump to 100

### PR DESCRIPTION
**Pull request by tr3mulant**
Changes:
- Added some SEO stuff (title and meta desc) to `components/HeadTag`
- Mocked some problematic libraries so we could resolve jest tests.
    - Intersection Observer
    - Framer Motion
    - Swiper
- Fixed the `useMedia` hook so that is it uses `useEffect` instead of `useEffectLayout` as it was causing hydration errors. Corrected a mistake that was returning early. 
- `yarn doc-dev` now brings down any running containers for the project and then tries to build 'n up them

### CODE REVIEW

File: __mocks__/framer-motion/framer-motion.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+		.fn()
+		.mockImplementation(({ children }) => children);
+
+	const FakeAnimatePresence = jest
+		.fn()
+		.mockImplementation(({ children }) => (
+			<FakeTransition>{children}</FakeTransition>
+		));
+
+	const motion = {
```
Comment (calebn) [inline __mocks__/framer-motion/framer-motion.js:12]: This is impressive but how did you know?

**Review by calebn — COMMENTED** [review]
Looks great, built your version and ran the tests with no errors.

File: __mocks__/framer-motion/framer-motion.js (hunk 1 above)
Comment (tr3mulant) [reply __mocks__/framer-motion/framer-motion.js:12]: Huh? What do you mean?

**Comment by calebn** [comment]
Meant to merge this earlier...

**Pull request merged**